From c4b76cd019581b84f250b838ce070687de27b7dd Mon Sep 17 00:00:00 2001
From: Thomas Orozco <thomas@orozco.fr>
Date: Wed, 1 Apr 2015 10:42:13 +0200
Subject: [PATCH] Add better error logging

---
 php_memcached_session.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/php_memcached_session.c b/php_memcached_session.c
index de8d165..bd509a8 100644
--- a/php_memcached_session.c
+++ b/php_memcached_session.c
@@ -72,7 +72,7 @@ static int php_memc_sess_lock(memcached_st *memc, const char *key TSRMLS_DC)
 				write_retry_attempts--;
 				continue;
 			}
-			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Write of lock failed");
+			php_error_docref(NULL TSRMLS_CC, E_WARNING, "Memcached: Write of lock failed: %s", memcached_strerror(memc, status));
 			break;
 		}
 
@@ -322,6 +322,7 @@ PS_READ_FUNC(memcached)
 		free(payload);
 		return SUCCESS;
 	} else {
+		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Memcached: Failed to read session data: %s", memcached_strerror(memc_sess->memc_sess, status));
 		return FAILURE;
 	}
 }
@@ -360,6 +361,7 @@ PS_WRITE_FUNC(memcached)
 		}
 	} while (write_try_attempts > 0);
 
+	php_error_docref(NULL TSRMLS_CC, E_WARNING, "Memcached: Failed to write session data, last error was: %s", memcached_strerror(memc_sess->memc_sess, status));
 	return FAILURE;
 }
 
-- 
2.2.2

